CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc

```diff
@@ -912,23 +1089,77 @@ void SiStripGainFromCalibTree::processEvent() {
 		//(Charge_Vs_Index_Absolute[elepos])->Fill(APV->Index,Charge);   
 		(Charge_Vs_Index[elepos])         ->Fill(APV->Index,ClusterChargeOverPath);
 
-		if(APV->SubDet==StripSubdetector::TIB){ (Charge_Vs_PathlengthTIB[elepos])  ->Fill((*path)[i],Charge); 
-		}else if(APV->SubDet==StripSubdetector::TOB){ (Charge_Vs_PathlengthTOB[elepos])  ->Fill((*path)[i],Charge);
-		}else if(APV->SubDet==StripSubdetector::TID){
-			if(APV->Eta<0){			  (Charge_Vs_PathlengthTIDM[elepos]) ->Fill((*path)[i],Charge);
-			}else if(APV->Eta>0){                      (Charge_Vs_PathlengthTIDP[elepos]) ->Fill((*path)[i],Charge);
-			}
-		}else if(APV->SubDet==StripSubdetector::TEC){
-			if(APV->Eta<0){
-				if(APV->Thickness<0.04){          (Charge_Vs_PathlengthTECM1[elepos])->Fill((*path)[i],Charge);
-				}else if(APV->Thickness>0.04){          (Charge_Vs_PathlengthTECM2[elepos])->Fill((*path)[i],Charge);
-				}
-			}else if(APV->Eta>0){
-				if(APV->Thickness<0.04){          (Charge_Vs_PathlengthTECP1[elepos])->Fill((*path)[i],Charge);
-				}else if(APV->Thickness>0.04){          (Charge_Vs_PathlengthTECP2[elepos])->Fill((*path)[i],Charge);
-				}
-			}
-		}
+
+                // Compute the charge for monitoring and fill the relative histograms
+                int mCharge1 = 0;
+                int mCharge2 = 0;
+                int mCharge3 = 0;
+                int mCharge4 = 0;
+                if(APV->SubDet>2) {
+                    for(unsigned int s=0;s<(*nstrips)[i];s++){
+                        int StripCharge =  (*amplitude)[FirstAmplitude-(*nstrips)[i]+s];
+                        if(StripCharge>1024)      StripCharge = 255;
+                        else if(StripCharge>254)  StripCharge = 254;
+                        mCharge1 += StripCharge;
+
+                        StripCharge =  (*amplitude)[FirstAmplitude-(*nstrips)[i]+s];
+                        if(StripCharge>1024)      StripCharge = 255;
+                        else if(StripCharge>254)  StripCharge = 254;
+                        mCharge2 += StripCharge;
+
+                        StripCharge =  (*amplitude)[FirstAmplitude-(*nstrips)[i]+s];
+                        if(StripCharge>1024)      StripCharge = 255;
+                        else if(StripCharge>254)  StripCharge = 254;
+                        mCharge3 += StripCharge;
+
+                        StripCharge =  (*amplitude)[FirstAmplitude-(*nstrips)[i]+s];
```

Comment:
sorry, I don't follow why this isn't the same calculation repeated 4 times. Is there an iterator I don't see?